### DISCUSSION TITLE
[webpack-dev-middleware] [Error: × TypeError: Cannot read properties of undefined (reading 'set')

### DISCUSSION THREAD

**Discussion by X-jc** [web-infra-dev/rsbuild#3516] [Q&A]
<e> [webpack-dev-middleware] [Error:   × TypeError: Cannot read properties of undefined (reading 'set')
<e>   │     at Object.fn (D:\project\yka\yka-admin-v2\node_modules\webpack\lib\DefinePlugin.js:359:37)
<e>   │     at SyncHook.callAsyncStageRange (D:\project\yka\yka-admin-v2\node_modules\@rspack\lite-tapable\dist\index.js:240:21)
<e>   │     at SyncHook.callStageRange (D:\project\yka\yka-admin-v2\node_modules\@rspack\lite-tapable\dist\index.js:259:14)
<e>   │     at QueriedHook.call (D:\project\yka\yka-admin-v2\node_modules\@rspack\lite-tapable\dist\index.js:218:26)
<e>   │     at D:\project\yka\yka-admin-v2\node_modules\@rspack\core\dist\Compiler.js:570:244
<e>   │     at last.function (D:\project\yka\yka-admin-v2\node_modules\@rspack\core\dist\Compiler.js:860:28)
<e>   │ 
<e> ] {
<e>   code: 'GenericFailure'
<e> }

vue3
vue-cli迁移rsbuild后报如上错误，不清楚是什么原因，求解

**Comment by 9aoy** [comment]
you should use [rspack.DefinePlugin](https://rspack.dev/plugins/webpack/define-plugin#defineplugin) or [rsbuild source.define](https://rsbuild.dev/config/source/define) configuration instead of `webpack.DefinePlugin`

**Reply by X-jc** [reply]
感谢你的答复，是否还有其他办法可以解决这个问题，或者兼容webpack.DefinePlugin这种写法
